homeassistant/components/modbus/base_platform.py

```diff
@@ -86,13 +91,35 @@ def __init__(self, hub: ModbusHub, entry: dict[str, Any]) -> None:
     async def async_update(self, now=None):
         """Virtual function to be overwritten."""
 
-    async def async_base_added_to_hass(self):
-        """Handle entity which will be added."""
+    @callback
+    def async_remote_start(self) -> None:
+        """Remote start entity."""
+        if self._cancel_timer:
+            self._cancel_timer()
+            self._cancel_timer = None
         if self._scan_interval > 0:
-            cancel_func = async_track_time_interval(
+            self._cancel_timer = async_track_time_interval(
                 self.hass, self.async_update, timedelta(seconds=self._scan_interval)
             )
-            self._hub.entity_timers.append(cancel_func)
+        self._attr_available = True
+        self.async_write_ha_state()
+
+    @callback
+    def async_remote_stop(self) -> None:
+        """Remote stop entity."""
+        if self._cancel_timer:
+            self._cancel_timer()
+            self._cancel_timer = None
+        self._attr_available = False
+        self.async_write_ha_state()
+
+    async def async_base_added_to_hass(self):
+        """Handle entity which will be added."""
+        self.async_remote_start()
+        async_dispatcher_connect(self.hass, SIGNAL_STOP_ENTITY, self.async_remote_stop)
```

Comment:
Side note: If we'd add support for removing the entity, we'd need to disconnect the signal when removing the entity, by using `self.async_on_remove` here.

Reply:
Thanks for the hint, and your review suggestions.